packages/components/ovh-shell/README.md

```diff
@@ -1 +1,178 @@
+  - `pluginId`: Your plugin `name`/`id`.
+  - `availabilty`: A boolean that sets the availability of a plugin to the desired value.
+
+### Plugins
+
+The lib exposes a set of plugins already ready to use, to save you the time of creating them for each app.
+Available plugins :
+
+- i18n
+- routing
+- auth
+- ux
+- tracking
+- navigation
+
+#### `i18n`
+
+- `getLocale`(`(shell: Shell, environment: Environment) => string`): Given a shell and an environment, returns the current locale of the user.
+- `setLocale`(`(locale: string) => void`): Sets new user locale based on given parameter.
+
+*Example :*
+
+```ts
+import { plugin } from '@ovh-ux/shell';
+
+const i18n = plugin.i18n(shell, environment);
+
+// Get locale
+const currentLocale = i18n.getLocale();
+
+// Set locale
+i18n.setLocale('en_GB');
+```
+
+#### `routing`
+
+- `initRouting`(`iframe: HTMLFrameElement`) : Initializes routing for the manager based on the URL in the iframe.
+
+*Example :*
+
+```ts
+import { plugin } from '@ovh-ux/shell';
+
+const routingPlugin = plugin.routing;
+
+routingPlugin.initRouting(iframe);
+```
+
+#### `auth`
+
+- `login()`: Login to the manager.
+- `logout()`: Logout from the manager.
+
+*Example :*
+
+```ts
+import { plugin } from '@ovh-ux/shell';
+
+const { auth } = plugin;
+
+auth.login();
+auth.logout();
+```
+
+#### `ux`
+
+This plugin handles everything that is UI related, such as elements shown or hidden, the state or position of certain elements etc.
+Currently, it handles the state of the account sidebar, notifications sidebar and verifies cookie verification for the ssoAuthModal.
+
+- `isAccountSidebarVisible`:(`() => boolean`) : Returns if Account Sidebar is shown or not.
+- `isNotificationsSidebarVisible`:(`() => boolean`)
+- `showAccountSidebar`:(`() => void`)
+- `hideAccountSidebar`:(`() => void`)
+- `enableAccountSidebarVisibilityToggle`:(`() => void`) : Allow the visibility of the sidebar to be toggled. If it is disabled, you won't be able to open or close the sidebar.
+- `disableAccountSidebarVisibilityToggle`:(`() => void`) : Same logic as the previous function.
+- `toggleNotificationsSidebarVisibility`:(`() => void`)
+- `toggleAccountSidebarVisibility`:(`() => void`)
+- `getUserIdCookie`: (`() => string`): Returns the latest value of `USER_ID` cookie.
+- `getSSOAuthModalMode`:(`(oldUserID: string) => string`): Returns the mode in which the SSOAuth Modal is in. Three possible values : 'CONNECTED_TO_DISCONNECTED', 'DISCONNECTED_TO_CONNECTED', 'CONNECTED_TO_OTHER'.
+
+*Example :*
+
+```ts
+import { plugin } from '@ovh-ux/shell';
+
+const { ux } = plugin;
+
+ux.showAccountSidebar();
+```
+
+#### `tracking`
+
+The tracking API exposes everything that the library `ovh-at-internet` exposes with the same signature.
+So basically, you can use all the functions exposed by `ovh-at-internet` the same way via this plugin.
```

Comment:
That's a good idea, but we don't have the documentation for that for now... Maybe it could be the time to write it too, but the lib is just a wrapper of an external lib that we use in private, so i don't know if we can expose their docs ?